roles/links-on-desktop/tasks/main.yml

```diff
@@ -1,14 +1,25 @@
 ---
-# Playbook to configure Postfix on a server to be able to send emails to admins
+# Role to Create Link-Folder on Users Desktops
+
+- name: Check if path exists
+  stat:
+    path: "{{ adfc_link_path }}"
+  register: stat_result
+
+- name: show results
+  debug:
+    msg: "Path {{ adfc_link_path }} exists"
+  when: stat_result.stat.exists == True
 
 - name: create links folder on UCS-Master
   file:
-    path: /adfc/links/
+    path: "{{ adfc_link_path }}"
     state: directory
     owner: root
     group: root
     mode: 0755
   delegate_to: ucs-master.gst.hamburg.adfc.de
+  when: stat_result.stat.exists == False
```

Comment:
Finde ich auch nicht besser. Im Moment braucht die setup-client Rolle schon 2 Minuten 53 Sekunden.  Der "create links folder on UCS-Master" prüft selber ob der Ordner schon existiert.
was hälst du von:
```
when: ansible_hostname == "ucs-master"
```
und unten bei der anderen Rolle ein:
```
 when: "'arbeitsplatz' in group_names"
```
dann kannst du die Rolle in setup-arbeitsplatz und setup-master aufnehmen

Reply:
klingt gut, 
ich versuche das mal

Reply:
ist implementiert und erfolgreich getestet.
Etwas tückisch bei der Reihenfolge der kombinierten Tests 

`when: ansible_hostname == "ucs-master" and stat_result.stat.exists` ==> funktioniert
vs. 
`when: stat_result.stat.exists and ansible_hostname == "ucs-master"` ==> funktioniert nicht, da "stat_result" vorher nicht definiert wurde

Reply:
Der Ordner /adfc/links/ wird übrigens -vermutlich durch den rsync job- regelmäßig gelöscht.

Reply:
Ja, aber vermutlich nicht mehr lange :-)